src/App.tsx

```diff
@@ -15,35 +16,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  SortByAlphabet = 'alphabet',
+  SortByLength = 'length',
+  None = '',
+}
+
+function getPrepearedGoods(
+  goods: string[],
+  { sortField, isReversed }: { sortField: SortType; isReversed: boolean },
+) {
+  const prepearedGoods = [...goods];
+
+  if (sortField) {
+    prepearedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.SortByAlphabet:
+          return good1.localeCompare(good2);
+        case SortType.SortByLength:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.None);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
```

Comment:
no needs to add generic here
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```